woocommerce/woocommerce-gutenberg-products-block
# Legacy Template: Add warning about block removal (#5207)

**For full context, read: #5180**

Legacy blocks were previous locked for removal to avoid unintended consequences. However, this would lock the ability to move the block within other e.g. layout blocks and unnecessarily limit merchant customization ability.

Now that we have [reverted this decision](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5176), merchants could delete this block, which is likely **not** what they want to do. While we investigate other, more long-term, solutions, we are adding a warning notice.

Refs #5180.


Fixes #5207

## Discussion

> ⚠️ This PR is a draft, we need to come up with the definitive copy

@vivialice I have consulted with @frontdevde and he mentioned you would be the best person to come up with some good copy for this block. I took some of your inputs from the conversation at #5180 and added them as a draft. But I'd be super thankful if you could come up with the best version of this :)

Also, [you have mentioned](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5180#issuecomment-972729497) adding some sort of link to troubleshooting docs. Me and @frontdevde tried to look for the best place for it in WooCommerce.com dashboard, but were not sure, so I'd love to ask your guidance here as well 🙏 

### Screenshots

*Before:*
<img width="383" alt="Screen Shot 2021-11-21 at 18 24 50" src="https://user-images.githubusercontent.com/1847066/142772474-850a4e50-ae99-408e-87f2-761e46f21319.png">

*After:*

![Screen Shot 2021-11-21 at 18 20 08](https://user-images.githubusercontent.com/1847066/142772481-53a6efa3-30d8-4224-9dee-01eaa32bb7fa.png)

### Testing

### Manual Testing

How to test the changes in this Pull Request:

1. Go to `Appearence → Editor`
2. Select `General templates → Product Archive Page`
3. See longer warning notice on the legacy block

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
